### DISCUSSION TITLE
Transferability of potentials on different periodic cells

### DISCUSSION THREAD

**Discussion by cedenoruel** [MarsalekGroup/aml#23] [Q&A] [answered]
Hello Developers,

I have a question on the transferability of the trained potentials.
For instance, if I trained using the (100)-face of a slab in aqueous environment, I wonder if the resulting C-NNP potentials would be valid for different faces (101), (110), etc which have different lattice parameters.

Looking at the codes, the cell parameter appears to be optional.

`cell: a constant cell to use in all frames, optional`
`cell: cell vectors, array-like of dimensions (3, 3)`

In order to include the cell parameters in the training, I imagine to include the `&CELL` in the trajectory section of the CP2K input file?

Also, would it be possible to train by combining the trajectory of different faces with different lattice parameters?

Thanks.

**Comment by cschran** [comment, marked as answer]
Dear cedenoruel,

thanks for your interest in the AML package.
In #12 there is an example of how you can add the cell information when reading in CP2K files.

Regarding your question about transferability, this depends quite a bit on your actual system of interest. In some cases, your model trained on one face might be working already for others and monitoring the committee disagreement will then give you an idea of how much more uncertain the model is in the new case.

Best is usually to include these other structures explicitly and your suggestion of running 3 different QbC jobs and combining the training set for the final model would be well suited.
Note that you can add different structures easily as they behave like lists:
```
structures = aml.Structures.from_file('../QbC1/iteration-015/train-000/input.data')
structures += aml.Structures.from_file('../QbC2/iteration-015/train-000/input.data')
```

**Reply by cedenoruel** [reply]
Great! Thanks @cschran  for the very detailed response. Looks very promising.

I have another unrelated question though.

By default, the training requires at least 2000 MD structures, which in my case, is computationally expensive at DFT level ( 600+ atoms and for multiple faces).  Is there a built-in functionality to do a Δ-ML approach, that is:
1. Train with 2000+ structures at the DFTB level (or lower)
2. Train a model that transforms the DFTB level energies to DFT energies (by some sort of scaling factor) using a smaller training set.

**Reply by cschran** [reply]
This is at the moment only supported in an indirect way, where you will need to provide the correct energies and forces of the difference between DFTB and DFT when reading in the structure.
This can be done e.g. using the reftraj feature in cp2k + a mixed force environment, where you subtract the DFTB result from the DFT result for every structure in your reftraj.

We're also working on a more flexible version of AML, which will hopefully be realised in the near future and would support adding a cp2k calculator during QbC.

**Reply by cedenoruel** [reply]
I'm looking forward to that. Thanks @cschran .
